src/channel_value.rs

```diff
@@ -40,7 +40,7 @@ impl clap::builder::TypedValueParser for ChannelValueParser {
         let (name, url) = buf.split_once('=').ok_or_else(|| {
             clap::Error::raw(
                 clap::error::ErrorKind::InvalidValue,
-                "Should be formatted `name=url`",
+                "`--channel` should be formatted `name=url`, eg `--channel nixpkgs=https://nixos.org/channels/nixpkgs-unstable`. To not use a channel, pass `--channel`",
```

Comment:
Really? `nix-installed [...] --channel` will not use a channel? Seems weird to me; I'd expect `--no-channel` or something.

Reply:
Yeah, really. That's what clap infers. Even if we added `--no-channel` this would still be valid, so it'd just be another flag. I should add something to `--help` though.

Reply:
I feel like there should almost be a way to disable that... Because --no-channel makes so much more sense.

(That said, the nix cli is fond of setting various settings to an empty string as a way of disabling them so this isn't particularly out of flavor here)